README.md

```diff
@@ -25,24 +24,37 @@ interface.
 
 This repository provides the following components of the Cylc system.
 
-* The UI
+* **The UI**
 
   This is the Cylc web app that provides control and monitoring functionalities
   for Cylc workflows.
 
-* The UI Server
+  > The UI is developed in a separate repository https://github.com/cylc/cylc-ui
+
+* **The UI Server**
 
   This is a web server which serves the Cylc web UI. It connects to running
   workflows and workflow databases to provide the information the UI displays.
+  It is a [Jupyter Server](https://github.com/jupyter-server/jupyter_server).
```

Comment:
A Jupyter Server application, or a Jupyter web Application? I'm reading their docs, from what I understood, Jupyter Server is the utility code for creating applications or extensions. Applications being complete web applications that use the Jupyter Server, and Extensions only extend the REST API.

Reply:
Huh, so looks like everything is an extension, but there's a special `ExtensionApp` that makes an extension that behaves like the old applications  (like notebook, and current UI Server I think?). So everything is an extension? :smile:

Reply:
So there's some similar but different terminology here which makes things confusing.

The Cylc UI Server (as of this PR) is a "Jupyter Server extension applicaiton".

* The "extension" part is because it is extending Jupyter Server.
* The application part is because it is a configurable traitlets application.

Extension applications can:

* Add static files (automatically served by the server)
  * Not doing this right now because of https://github.com/cylc/cylc-uiserver/issues/231
* Add handlers.
* Request resources from the server.

Jupyter server also offers non-configurable extensions (ones which aren't applications), these are cruder.